test/cli/commands_test.py

```diff
@@ -1012,6 +1014,12 @@ def test__cli__command_lint_serialize_multiple_files(serialize, write_file, tmp_
         result = json.loads(result_payload)
         filepaths = {r["file"] for r in result}
         assert len(filepaths) == 1
+    elif serialize == "github-annotation-native":
+        result = result_payload.split("\n")
+        # SQLFluff produces trailing newline
+        if result[-1] == "":
+            del result[-1]
+        assert len(result) == 24
```

Comment:
What is this here for? Shouldn't it be variable based on the number of results?

Reply:
I wasn't entirely sure either. But the trailing new line behaves differently between "write to file" and reading the raw output. The raw output has trailing newline while the file has not.

However, looking at the code again, I might have misunderstood what this test is trying to ... test 😂

Reply:
I'm OK with removing the trailing line. I'm more questioning why it should be exactly 24 lines long?

Reply:
The input file throws 12 linting errors. So, with 24 we are checking that the file actually gets linted twice. But I will try to refactor this a bit to make it more intuitive

Reply:
This test lints the same input file into various output formats. So this line is only executed once, when the file is linted into `"github-annotation-native"` format. (It's probably confusing because here we're using `parametrize` to vary the _format_ rather than the _input file_.

Reply:
Ah I missed this was a test. Ok hard coding the expected results makes total sense then. But yeah maybe a brief investigation to see why it’s doing it twice and if that can be avoided and, if not, a comment to explain.

Reply:
Oh, just noticed the part about twice. That's really weird, and worth investigating. Does the same thing happen if you run the same command manually (from the terminal)?

Reply:
I was working on this same test for another PR (#3109) and noticed that it is specifically designed to process the same file twice. Kinda weird test, but nothing wrong with this PR!

See these lines in the test:
```
    # note the file is in here twice. two files = two payloads.
    result = invoke_assert_code(
        args=[lint, cmd_args],
        ret_code=65,
    )
```